integration/container/nat_test.go

```diff
@@ -29,7 +29,15 @@ func TestNetworkNat(t *testing.T) {
 	startServerContainer(ctx, t, msg, 8080)
 
 	endpoint := getExternalAddress(t)
-	conn, err := net.Dial("tcp", net.JoinHostPort(endpoint.String(), "8080"))
+	var err error
+	var conn net.Conn
+	for retry := 0; retry < 10; retry++ {
+		conn, err = net.Dial("tcp", net.JoinHostPort(endpoint.String(), "8080"))
+		if err == nil {
+			break
+		}
+		time.Sleep(200 * time.Millisecond)
+	}
```

Comment:
I think these fixes didn't make much of a difference and I ended up skipping the test, as there was an issue open for 4 years and some other tests were skipped.

I pushed the branch here: https://github.com/rata/moby/tree/runc-1.2-test-fixes. See the last commit, that added a SKIP and mentions the issue. There are other test in this file that use it too.

Maybe we can just drop the retries in this file, just keep the chunk about waiting for the container to be running?